### DISCUSSION TITLE
enum descriptions?

### DISCUSSION THREAD

**Discussion by mszabo12** [SystemRDL/systemrdl-compiler#156] [General]
So this is probably a dumb question but the compiler documentation mentions UserEnum's have a rdl_desc property.  How would you actually set a description in an enum?  

The obvious method would be to use something like:
enum mode{
     	     desc = "This enum describes various AES key sizes";
	      mode0=2'b00    {desc = "runs mode 1";};
	      mode1=2'b01    {desc = "runs mode 2";};
	      mode2=2'b10    {desc = "runs mode 3";};
	      undefined=2'b11 {desc = "Undefined";};
	      };
However the compiler/language will treat 'desc' as the first encoding in the enum, and not as as a property.  I could see a description here being useful, but not sure how the language supports it.

**Comment by amykyta3** [comment]
The SystemRDL language does not support assigning properties to the enumerated type itself. This would include the `desc` property.

Looking at the spec's BNF grammar for enumerations, it makes it pretty clear (`enum_body` can only contain one or more `enum_entry` statements)

```
enum_def ::= enum id enum_body ;
enum_body ::= { enum_entry { enum_entry } }
enum_entry ::= id [ = constant_expression ] [ enum_property_assignment ] ;
enum_property_assignment ::= { { explicit_prop_assignment ; } }
```

**Reply by mszabo12** [reply]
That was certainly my thought just wondering if I had missed something since the UserEnum class has a rdl_desc property.  Would be useful to have a description here to have a nice block comment describing the enum in my generated C code.

**Comment by muneebullashariff** [comment]
Is there an example as how to specify the enum values in the .rdl file?

**Reply by amykyta3** [reply]
I have added a section about enums to the RDL tutorial that should provide a good example. See: https://peakrdl.readthedocs.io/en/latest/systemrdl-tutorial.html#enumerated-fields

Also, section 6.2.5 of the RDL spec goes into more details about enumerations: https://accellera.org/downloads/standards/systemrdl
